shell/platform/tizen/channels/text_input_channel.cc

```diff
@@ -192,7 +194,23 @@ void TextInputChannel::HandleMethodCall(
       if (input_type_json != input_type_info_json->value.MemberEnd() &&
           input_type_json->value.IsString()) {
         input_type_ = input_type_json->value.GetString();
+        bool is_signed = false;
+        auto is_signed_iter =
+            input_type_info_json->value.FindMember(kTextInputTypeSigned);
+        if (is_signed_iter != input_type_info_json->value.MemberEnd() &&
+            is_signed_iter->value.IsBool()) {
+          is_signed = is_signed_iter->value.GetBool();
+        }
+        bool is_decimal = false;
+        auto is_decimal_iter =
+            input_type_info_json->value.FindMember(kTextInputTypeDecimal);
+        if (is_decimal_iter != input_type_info_json->value.MemberEnd() &&
+            is_decimal_iter->value.IsBool()) {
+          is_decimal = is_decimal_iter->value.GetBool();
+        }
         input_method_context_->SetInputPannelLayout(input_type_);
+        input_method_context_->SetInputPanelLayoutVariation(is_signed,
+                                                            is_decimal);
       }
```

Comment:
I haven't tried it.. 
How about hiding and showing the panel to apply the layout change? 
```suggestion
ecore_imf_context_focus_out(imf_context_);
ecore_imf_context_input_panel_hide(imf_context_);

ecore_imf_context_focus_in(imf_context_);
ecore_imf_context_input_panel_show(imf_context_);
```

Reply:
Wow. I tried this solution but it only worked on mobile devices (TM1) but not on watch devices. I'll apply the change anyway.

Reply:
It turned out that the issue is only reproducible on my SM-R800 (Tizen 5.5) but not all watch devices. The solution works fine on watch emulators (5.5 and 6.0).